        Patches are taken from stlinux23-target-util-linux-2.12r-14.src.rpm
        (see ftp.stlinux.com)
--- util-linux-2.10q.orig/debian/hwclock.sh	Thu Mar 22 22:43:25 2001
+++ util-linux-2.10q/debian/hwclock.sh	Thu Mar 22 22:53:56 2001
@@ -15,6 +15,9 @@
 # WARNING:	Please read /usr/share/doc/util-linux/README.Debian.hwclock
 #		before changing this file. You risk serious clock
 #		misbehaviour otherwise.
+#
+# chkconfig: S 50 25
+#
 
 . /etc/default/rcS
 [ "$GMT" = "-u" ] && UTC="yes"
--- util-linux-2.10q/login-utils/agetty.c~	Sat Apr 21 14:23:41 2001
+++ util-linux-2.10q/login-utils/agetty.c	Sat Apr 21 14:25:55 2001
@@ -178,6 +178,8 @@
     int     code;
 };
 
+int noBaudGiven = 0;
+
 static struct Speedtab speedtab[] = {
     { 50, B50 },
     { 75, B75 },
@@ -285,7 +287,7 @@
 #endif
 
 #ifdef DEBUGGING
-	dbf = fopen("/dev/ttyp0", "w");
+	dbf = fopen("/dev/tty1", "w");
 
 	{	int i;
 	
@@ -479,18 +481,24 @@
 	}
     }
  	debug(_("after getopt loop\n"));
-    if (argc < optind + 2)			/* check parameter count */
+    if (argc < optind + 1)			/* check parameter count */
 	usage();
-
-    /* we loosen up a bit and accept both "baudrate tty" and "tty baudrate" */
-    if('0' <= argv[optind][0] && argv[optind][0] <= '9') {
-	/* a number first, assume it's a speed (BSD style) */
-	parse_speeds(op, argv[optind++]);	/* baud rate(s) */
-	op->tty = argv[optind];			/* tty name */
-    } else {
-	op->tty = argv[optind++];		/* tty name */
-	parse_speeds(op, argv[optind]);		/* baud rate(s) */
+    else if( argc < optind + 2)
+    {
+	noBaudGiven = 1;
+	op->tty =argv[optind];
     }
+    else{
+    /* we loosen up a bit and accept both "baudrate tty" and "tty baudrate" */
+    	if('0' <= argv[optind][0] && argv[optind][0] <= '9') {
+		/* a number first, assume it's a speed (BSD style) */
+		parse_speeds(op, argv[optind++]);	/* baud rate(s) */
+		op->tty = argv[optind];			/* tty name */
+	    } else {
+		op->tty = argv[optind++];		/* tty name */
+		parse_speeds(op, argv[optind]);		/* baud rate(s) */
+	    }
+      }
 
     optind++;
     if (argc > optind && argv[optind])
@@ -689,14 +697,6 @@
      * protections. SunOS 4 login will change the protections to 0620 (write
      * access for group tty) after the login has succeeded.
      */
-
-    /*
-     * Let us use 0600 for Linux for the period between getty and login
-     */
-
-    (void) chown(tty, 0, 0);			/* root, sys */
-    (void) chmod(tty, 0600);			/* 0622: crw--w--w- */
-    errno = 0;					/* ignore above errors */
 }
 
 /* termio_init - initialize termio settings */
@@ -722,7 +722,11 @@
     (void) ioctl(0, TCFLSH, TCIOFLUSH);
 #endif
 
-    tp->c_cflag = CS8 | HUPCL | CREAD | speed;
+    if(!noBaudGiven)
+    	tp->c_cflag = CS8 | HUPCL | CREAD | speed;
+    else
+        tp->c_cflag = CS8 | HUPCL | CREAD | tp->c_cflag;
+
     if (op->flags & F_LOCAL) {
 	tp->c_cflag |= CLOCAL;
     }
# When building swpaon, a script is run to determine
# details of the target system, and this generates header files with the
# correct defines. Unfortunatly this doesn't take account of cross
# compilation, so generate the file correctly here.

--- util-linux-2.11z.orig/mount/swapargs.h	1970-01-01 01:00:00.000000000 +0100
+++ util-linux-2.11z/mount/swapargs.h	2003-09-16 12:07:03.000000000 +0100
@@ -0,0 +1,3 @@
+#define SWAPON_HAS_TWO_ARGS
+#include <asm/page.h>
+#include <sys/swap.h>
diff -Naur util-linux-2.12m.orig/fdisk/fdiskbsdlabel.h util-linux-2.12m/fdisk/fdiskbsdlabel.h
--- util-linux-2.12m.orig/fdisk/fdiskbsdlabel.h	2005-02-22 13:51:44.467166976 +0000
+++ util-linux-2.12m/fdisk/fdiskbsdlabel.h	2005-02-22 13:52:42.215387912 +0000
@@ -43,7 +43,8 @@
 
 #define BSD_LINUX_BOOTDIR "/usr/ucb/mdec"
 
-#if defined (i386) || defined (__sparc__) || defined (__arm__) || defined (__mips__) || defined (__s390__) || defined (__sh__) || defined(__x86_64__)
+#if defined (i386) || defined (__sparc__) || defined (__arm__) || defined (__mips__) \
+|| defined (__s390__) || defined (__sh__) || defined(__x86_64__) || defined (__st200__)
 #define BSD_LABELSECTOR   1
 #define BSD_LABELOFFSET   0
 #elif defined (__alpha__) || defined (__powerpc__) || defined (__ia64__) || defined (__hppa__)
diff -Naur util-linux-2.12m.orig/fdisk/llseek.c util-linux-2.12m/fdisk/llseek.c
--- util-linux-2.12m.orig/fdisk/llseek.c	2005-02-22 13:51:44.468166824 +0000
+++ util-linux-2.12m/fdisk/llseek.c	2005-02-22 13:52:13.285785880 +0000
@@ -19,7 +19,7 @@
 
 #else	/* HAVE_LLSEEK */
 
-#if defined(__alpha__) || defined(__ia64__) || defined(__s390x__)
+#if defined(__alpha__) || defined(__ia64__) || defined(__s390x__) || defined(__st200__)
 
 #define my_llseek lseek
 
diff -Naur util-linux-2.12m.orig/fdisk/sfdisk.c util-linux-2.12m/fdisk/sfdisk.c
--- util-linux-2.12m.orig/fdisk/sfdisk.c	2005-02-22 13:51:44.468166824 +0000
+++ util-linux-2.12m/fdisk/sfdisk.c	2005-02-22 13:52:22.273419552 +0000
@@ -172,7 +172,7 @@
  * Note: we use 512-byte sectors here, irrespective of the hardware ss.
  */
 #undef use_lseek
-#if defined (__alpha__) || defined (__ia64__) || defined (__x86_64__) || defined (__s390x__)
+#if defined (__alpha__) || defined (__ia64__) || defined (__x86_64__) || defined (__s390x__) || defined (__st200__)
 #define use_lseek
 #endif
 
diff -Naur util-linux-2.12m/disk-utils/fsck.cramfs.c util-linux-2.12m-stm/disk-utils/fsck.cramfs.c
--- util-linux-2.12m/disk-utils/fsck.cramfs.c	2004-12-11 14:53:16.000000000 +0000
+++ util-linux-2.12m-stm/disk-utils/fsck.cramfs.c	2005-03-08 11:36:41.274888600 +0000
@@ -77,6 +77,9 @@
 #define PAD_SIZE 512
 
 #include <asm/page.h>
+#if defined __sh__ || defined __arm__
+#define PAGE_CACHE_SIZE (4096)
+#else
 #ifdef PAGE_SIZE
 #define PAGE_CACHE_SIZE ((int) PAGE_SIZE)
 #elif defined __ia64__
@@ -86,6 +89,7 @@
 #else
 #define PAGE_CACHE_SIZE (4096)
 #endif
+#endif
 
 /* Guarantee access to at least 8kB at a time */
 #define ROMBUFFER_BITS	13
--- util-linux-2.12r/mount/nfs4mount.c.orig	2006-11-01 17:07:14.000000000 +0000
+++ util-linux-2.12r/mount/nfs4mount.c	2006-11-01 17:07:47.000000000 +0000
@@ -37,6 +37,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <rpc/auth.h>
+#include <rpc/pmap_clnt.h>
 #ifdef HAVE_rpcsvc_nfs_prot_h
 #include <rpcsvc/nfs_prot.h>
 #else
